airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java

```diff
@@ -10,6 +10,8 @@
 
 public class SnowflakeDestination extends SwitchingDestination<SnowflakeDestination.DestinationType> {
 
+  public static boolean isAlive; // used for refresh Oauth token thread
```

Comment:
I'm not a fan of using global variables to keep state. Could we pass an `AtomicBoolean` to `DbConfigurationLoader` instead?

Reply:
Hi @girarda.
Performed additional investigation, but the problem is that this connector has lot's of other uploading types. Each of them uses different abstractions. The snowflake-destination is also inherited from other abstractions. So it will affect dozens of connectors. There are also no place that is called at the end of sync in all of the cases other than just main() method. A lot of code would require updates. So I will have to manually check all of the cases like check connections(for all types of uploading types, all of sync types etc), i.e. pretty easy to miss something. If that's the case what would you say if I propose some kind in singleton object that would work like a "state manager"?  
Thanks

Reply:
Updated, thanks